t/python/t0034-queuelist.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+###############################################################
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import math
+import unittest
+
+try:
+    import tomllib  # novermin
+except ModuleNotFoundError:
+    from flux.utils import tomli as tomllib
+
+import flux
+from flux.queue import QueueList
+from subflux import rerun_under_flux
+
+
+def __flux_size():
+    return 2
+
+
+class TestQueueList(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        self.fh = flux.Flux()
+
+    def test_001_anonymous(self):
+        qlist = QueueList(self.fh)
+        self.assertEqual(len(list(qlist)), 1)
+        queue = qlist[""]
+        self.assertEqual(queue.name, "")
+        self.assertTrue(queue.enabled)
+        self.assertEqual(queue.started, True)
+        self.assertEqual(queue.limits.duration, math.inf)
+        self.assertEqual(queue.limits.timelimit, math.inf)
+        self.assertEqual(queue.limits.max.nnodes, math.inf)
+        self.assertEqual(queue.limits.min.nnodes, 0)
+        self.assertEqual(queue.is_default, True)
+        self.assertEqual(queue.resources.all.nnodes, 2)
+        self.assertEqual(queue.resources.up.nnodes, 2)
+        self.assertEqual(queue.resources.free.nnodes, 2)
+        self.assertEqual(queue.resources.allocated.nnodes, 0)
+
+    def test_002_named(self):
+        testconf = """
+        [queues.batch]
+
+        [queues.debug]
+        policy.limits.duration = "1h"
+        policy.limits.job-size.max.nnodes = 1
+        policy.limits.job-size.max.ncores = 2
+
+        [policy]
+        limits.duration = "24h"
+        jobspec.defaults.system.queue = "batch"
+        jobspec.defaults.system.duration = "8h"
+        """
+        self.fh.rpc("config.load", tomllib.loads(testconf)).get()
+
+        qlist = QueueList(self.fh)
```

Comment:
Maybe add a test like `QueueList(self.fh, [])`?

Reply:
and also requesting specific queues, `QueueList(self.fh, ["batch"]`, and probably multiple queues `["batch", "debug"]`